controllers/hostedcontrolplane/healthcheck.go

```diff
@@ -0,0 +1,198 @@
+	if err != nil {
+		return fmt.Errorf("failed to increment healthcheck success count: %w", err)
+	}
+
+	successes = healthcheckConfigMapSuccesses(healthcheckConfigMap)
+	if successes > consecutiveSuccessfulHealthchecks {
+		return nil
+	}
+
+	if checkClusterOver1Hour(hostedcontrolplane.ObjectMeta.CreationTimestamp) {
+		return nil
+	}
+
+	return fmt.Errorf("insufficient successful health check attempts")
+}
+
+// getHealthCheckConfigMap retrieves the healthcheck configmap for the provided HCP from the cluster
+func (r *HostedControlPlaneReconciler) getHealthCheckConfigMap(ctx context.Context, hostedcontrolplane *hypershiftv1beta1.HostedControlPlane) (corev1.ConfigMap, error) {
+	configmap := buildHealthCheckConfigMap(hostedcontrolplane)
+	err := r.Get(ctx, types.NamespacedName{Name: configmap.Name, Namespace: configmap.Namespace}, &configmap)
+	return configmap, err
+}
+
+// createHealthcheckConfigMap creates a new configmap to track the healthchecking history of the provided HCP, and returns the resulting object along with any error encountered
+func (r *HostedControlPlaneReconciler) createHealthcheckConfigMap(ctx context.Context, hostedcontrolplane *hypershiftv1beta1.HostedControlPlane) (corev1.ConfigMap, error) {
+	configmap := buildHealthCheckConfigMap(hostedcontrolplane)
+	err := r.Create(ctx, &configmap)
+	return configmap, err
+}
+
+// buildHealthCheckConfigMap creates an empty configmap used to track the healthcheck history for a hostedcontrolplane cluster
+func buildHealthCheckConfigMap(hostedcontrolplane *hypershiftv1beta1.HostedControlPlane) corev1.ConfigMap {
+	configmap := corev1.ConfigMap{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:            fmt.Sprintf("%s-kube-apiserver-rmo-healthcheck", hostedcontrolplane.Name),
+			Namespace:       hostedcontrolplane.Namespace,
+			OwnerReferences: buildOwnerReferences(hostedcontrolplane),
+		},
+	}
+	return configmap
+}
+
+// healthcheckConfigMapSuccesses returns the number of recorded successful healthchecks the configmap has tallied under it's healthcheckAnnotation annotation.
+//
+// If the proper annotation cannot be found, or does not have an integer as its key, the success count is assumed to be 0
+func healthcheckConfigMapSuccesses(configmap corev1.ConfigMap) int {
+	value, found := configmap.Annotations[healthcheckAnnotation]
+	if !found {
+		// if the annotation hasn't been added yet, then healthcheck success count is 0
+		return 0
+	}
+
+	successes, err := strconv.Atoi(value)
+	if err != nil {
+		// if there's an invalid value on the annotation, just assume healthcheck success count is 0
+		return 0
+	}
+	return successes
+}
+
+// resetHealthCheckSuccesses sets the value of the healthcheck success counter to 0 on the configmap on-cluster, and returns an updated copy of the configmap
+func (r *HostedControlPlaneReconciler) resetHealthCheckSuccesses(ctx context.Context, configmap corev1.ConfigMap) (corev1.ConfigMap, error) {
+	delete(configmap.Annotations, healthcheckAnnotation)
```

Comment:
Is there any risk that we could try to delete an annotation that doesn't exist? do we need to handle such error?